handbook/engineering/index.md

```diff
@@ -82,6 +82,12 @@ Sourcegraph has a lot of repositories!
 - [Infrastructure repositories](https://github.com/sourcegraph?utf8=%E2%9C%93&q=repo-type-infrastructure&type=&language=)
 - [Customer infrastructure repositories](https://github.com/sourcegraph?utf8=%E2%9C%93&q=repo-type-infrastructure+repo-type-customer&type=&language=)
 
+## Where Sourcegraph is deployed:
```

Comment:
This heading doesn't really apply to the content. For instance, the Design System is not Sourcegraph. So many something a bit more generic is appropriate here. 

Also, I don't see the QA/Staging instance link here?

Reply:
Added more instances and changed title as we discussed in Slack.